src/plugins/openmeteo/datasource.py

```
@@ -0,0 +1,333 @@
+
+
+class OpenMeteoDataSource(DataSource):
+    def __init__(
+        self,
+        api_client: Optional[RateLimitedAPIClient] = None,
+        cache: Optional[Cache] = None,
+        metrics: Optional[MetricsCollector] = None,
+        base_url: str = "https://api.open-meteo.com/v1"
+    ):
+        self.base_url = base_url
+        self.api_client = api_client or RateLimitedAPIClient(base_url=self.base_url, rate_limit=10000)
```

Pylint (3.3.7)

[convention] Line too long (102/100)

(C0301)